src/app/Library/CrudPanel/Traits/Relationships.php

```diff
@@ -358,4 +408,132 @@ private function modelMethodIsRelationship($model, $method)
+        }
+
+        if (is_a($key, 'Illuminate\Database\Eloquent\Model', true)) {
+            if (in_array($key, $morphMap)) {
+                $key = $morphMap[array_search($key, $morphMap)];
+
+                if (array_key_exists($key, $morphTypeField['options'])) {
+                    throw new \Exception('Duplicate entry for «'.$key.'» in morphOptions');
+                }
+            }
+            $morphTypeField['options'][$key] = $label ?? Str::afterLast($key, '\\');
+        } else {
+            if (! array_key_exists($key, $morphMap)) {
+                throw new \Exception('Unknown morph type «'.$key.'», either the class doesnt exists, or the name was not found in the morphMap');
+            }
+
+            if (array_key_exists($key, $morphTypeField['options'])) {
+                throw new \Exception('Duplicate entry for «'.$key.'» in morphOptions');
+            }
+
+            $morphTypeField['options'][$key] = $label ?? ucfirst($key);
+        }
+
+        $morphIdField['morphOptions'][$key] = $options;
+
+        $morphTypeField = isset($morphField['morphTypeField']) ? array_merge($morphTypeField, $morphField['morphTypeField']) : $morphTypeField;
+        $morphIdField = isset($morphField['morphIdField']) ? array_merge($morphIdField, $morphField['morphIdField']) : $morphIdField;
+
+        $morphField['subfields'] = [$morphTypeField, $morphIdField];
+
+        if ($this->fields()[$fieldName] ?? false) {
+            $this->modifyField($fieldName, $morphField);
+        } else {
+            return $morphField;
+        }
+    }
+
+    /**
+     * Returns the morphable_id field structure for morphTo relations.
+     *
+     * @param  string  $relationName
+     * @param  string  $morphIdFieldName
+     * @return array
+     */
+    private static function getMorphidFieldStructure($relationName, $morphIdFieldName, $morphTypeFieldName)
+    {
+        return [
+            'name' => $morphIdFieldName,
+            'type' => 'relationship.morphTo_select',
+            'entity' => false,
+            'placeholder' => 'Select the '.$relationName,
+            'allows_null' => true,
+            'allow_multiple' => false,
+            'morphTypeFieldName' => $morphTypeFieldName,
+            'attributes' => [
+                'data-morph-select' => $relationName.'-morph-select',
+            ],
+            'wrapper' => ['class' => 'form-group col-md-9'],
+        ];
+    }
+
+    /**
+     * Returns the morphable_type field structure for morphTo relations.
+     *
+     * @param  string  $relationName
+     * @param  string  $morphTypeFieldName
+     * @return array
+     */
+    private static function getMorphTypeFieldStructure($relationName, $morphTypeFieldName)
+    {
+        return [
+            'name' => $morphTypeFieldName,
+            'type' => 'relationship.morphTo_type_select',
+            'placeholder' => 'Select the '.$relationName,
```

Comment:
I don't think it makes sense for the admin to write out the relation name. It will look super-weird to him. Yes, the developer might know what a `commentable` is... but the admin does not:

![CleanShot 2022-08-29 at 19 14 38](https://user-images.githubusercontent.com/1032474/187246676-c0e7f131-cb60-42b7-b805-c7b1edfab53c.png)


The developer (if they're any good) should hide that abstractization from the admin. I think the solution here is simple:

```suggestion
            'placeholder' => 'Select an entry',
```